ucp/benchmarks/asyncssh.py

```diff
@@ -0,0 +1,178 @@
+import asyncio
+import json
+import logging
+import os
+import queue
+import sys
+from functools import partial
+
+logger = logging.getLogger("ucx.asyncssh")
+logger.setLevel(logging.getLevelName(os.getenv("UCXPY_ASYNCSSH_LOG_LEVEL", "WARNING")))
+
+try:
+    import asyncssh
+
+    class SSHProc(asyncssh.SSHClientSession):
+        def __init__(self, out_queue):
+            assert isinstance(out_queue, queue.Queue)
+            self.out_queue = out_queue
+
+        def data_received(self, data, datatype):
+            logger.debug(f"SSHProc.data_received(): {data=}")
+            self.out_queue.put(data)
+
+        def connection_lost(self, exc):
+            if exc:
+                logger.error(f"SSH session error: {exc}", file=sys.stderr)
+            else:
+                logger.debug(
+                    f"SSH connection terminated succesfully {self.out_queue.empty()=}"
+                )
+
+    class SSHServerProc(SSHProc):
+        address = None
+        port = None
+
+        def data_received(self, data, datatype):
+            if self.address is None and self.port is None:
+                logger.debug(f"SSHServerProc.data_received() address: {data=}")
+
+                server_info = json.loads(data)
+                self.address = server_info["address"]
+                self.port = server_info["port"]
+
+                self.out_queue.put(server_info)
+            else:
+                super().data_received(data, datatype)
+
+    async def _run_ssh_cluster(
+        args,
+        server_host,
+        worker_hosts,
+        num_workers,
+        get_server_command,
+        get_worker_command,
+    ):
+        """
+        Run `ucp.benchmarks.cudf_merge` in an SSH cluster.
```

Comment:
This is no longer the case, right? It runs whatever command is specified?

Reply:
But the command is actually running `ucp.benchmarks.cudf_merge`, the difference is the code path it takes based on the arguments that get passed to it.

Reply:
For clarity, the commands that run:

https://github.com/rapidsai/ucx-py/pull/870/files#diff-df6376e50febb20e73957efbd9a7d24ca626cc0ba1bf32320848bb21d1978b5bR200

https://github.com/rapidsai/ucx-py/pull/870/files#diff-df6376e50febb20e73957efbd9a7d24ca626cc0ba1bf32320848bb21d1978b5bR228

Reply:
I guess what I meant was that this seems to be generic functionality. Right now the only caller is the `cudf_merge` benchmark, but if I were to provide alternative `get_server_command ` and `get_worker_command` functions, everything would work correctly, right? (e.g. suppose there were also a transpose benchmark, we could pass in appropriate callables that ended up running `ucp.benchmarks.transpose`).

Reply:
Ah yes, indeed that is intended for generic functionality, although not supported by other benchmarks at the moment. I attempted to clarify that now in https://github.com/rapidsai/ucx-py/pull/870/commits/18cac5cb108463ecf008e9bb6facf09e9805f684 .

Reply:
Thanks.